### PR TITLE
Develop to Main

### PR DESCRIPTION
**Pull request by minccccc**
Develop to Main

### CODE REVIEW

File: src/Warehouse/Application/Background/ProductsSyncService.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using MediatR;
+using Microsoft.Extensions.Options;
+using Quartz;
+
+namespace Application.Background
+{
+    public class ProductsSyncService : IJob
+    {
+        private readonly IMediator _mediator;
+
```
Comment (Admiraculix) [inline src/Warehouse/Application/Background/ProductsSyncService.cs]: Remove this new line

File: src/Warehouse/Application/Background/ProductsSyncService.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using Application.Configuration;
+using Application.Features.Products.Synchronize;
+using MediatR;
+using Microsoft.Extensions.Options;
+using Quartz;
+
+namespace Application.Background
+{
```
Comment (Admiraculix) [inline src/Warehouse/Application/Background/ProductsSyncService.cs]: Since we are using the latest framework let's use this approach for all files. This comment applies to all files:
[learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces)

also is configurable in VS2022 as option to be default.

File: src/Warehouse/Application/Configuration/AddFluentValidatorConfigExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using FluentValidation;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Application.Configuration
```
Comment (Admiraculix) [inline src/Warehouse/Application/Configuration/AddFluentValidatorConfigExtensions.cs]: This is an Extension method of IServiceCollection and this mean should be in folder Extensions/DependencyInjection or just Extensions.

File: src/Warehouse/Application/Configuration/AddFluentValidatorConfigExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public static class AddFluentValidatorConfigExtensions
```
Comment (Admiraculix) [inline src/Warehouse/Application/Configuration/AddFluentValidatorConfigExtensions.cs]: I can see the idea behind having them separated each to be a class with a method. But I don't understand what we gain from this. This would be suitable for common assemblies, where I have different functionalities in different folders, and if we want to use them, we register exactly that one.
Why don't we earn anything because our Application layer can't work without some of these technologies it relies on. For this, I believe that the following would be an appropriate approach:

Class name DependencyRegistrationExtention.cs
and Method name for you whole assembly  AddApplication, RegisterApplicationLayer or something like that.
And all you other registrations should be part of this method.

Only Quartz can be in separate private method in  our DependencyRegistrationExtention.cs because has soo configuration there.  it will be good to be in separate method AddQuartz or something like that.

File: src/Warehouse/Application/Configuration/MediatrConfigurationExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using Application.PipelineBehaviors;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Application.Configuration
+{
+    public static class MediatrConfigurationExtensions
+    {
+        public static void AddMediatrConfiguration(this IServiceCollection services)
```
Comment (Admiraculix) [inline src/Warehouse/Application/Configuration/MediatrConfigurationExtensions.cs]: Check previous commentary related to the topic.

File: src/Warehouse/Application/Configuration/QuartzConfigurationExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
+﻿using Application.Background;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Quartz;
+
+namespace Application.Configuration
+{
+    public static class QuartzConfigurationExtensions
+    {
+        public static void AddQuartzConfiguration(this IServiceCollection services, ConfigurationManager config)
```
Comment (Admiraculix) [inline src/Warehouse/Application/Configuration/QuartzConfigurationExtensions.cs]: Check previous commentary related to the topic.

File: src/Warehouse/Application/Configuration/ProductsSourceConfig.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Application.Configuration
+{
+    public class ProductsSourceConfig
+    {
+        public string Uri { get; set; }
```
Comment (Admiraculix) [inline src/Warehouse/Application/Configuration/ProductsSourceConfig.cs]: Because our application relies on this this configuration is always populated let's use the new access modifier from .net core 8: 
public -> required

File: src/Warehouse/Application/Configuration/ProductsSourceConfig.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+
+        public int RefreshTime { get; set; }
```
Comment (Admiraculix) [inline src/Warehouse/Application/Configuration/ProductsSourceConfig.cs]: public -> required

File: src/Warehouse/Application/Configuration/ProductsSourceConfig.cs [hunk 9]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Application.Configuration
```
Comment (Admiraculix) [inline src/Warehouse/Application/Configuration/ProductsSourceConfig.cs]: This file is configuration, it is correct place where exist, don't move with extension refactoring

File: src/Warehouse/Application/AppConstants.cs [hunk 10]
```diff
@@ -0,0 +1,21 @@
+﻿namespace Application
```
Comment (Admiraculix) [inline src/Warehouse/Application/AppConstants.cs]: It will be a good if you put this file into folder Constants

File: src/Warehouse/Application/Background/ProductsSyncService.cs [hunk 11]
```diff
@@ -0,0 +1,32 @@
+﻿using Application.Configuration;
+using Application.Features.Products.Synchronize;
+using MediatR;
+using Microsoft.Extensions.Options;
+using Quartz;
+
+namespace Application.Background
```
Comment (Admiraculix) [inline src/Warehouse/Application/Background/ProductsSyncService.cs]: Lets be moved  in folder Services/Backgrounds or Services/Jobs

File: src/Warehouse/Application/Background/ProductsSyncService.cs [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+            _productsSourceConfig = productsSourcs.Value;
+        }
+
+        public async Task Execute(IJobExecutionContext context)
+        {
+            try
+            {
+                await _mediator.Send(new SyncProductsCommand(_productsSourceConfig.Uri));
+            }
+            catch { }
```
Comment (Admiraculix) [inline src/Warehouse/Application/Background/ProductsSyncService.cs]: I think should catch and log

File: src/Warehouse/Application/Extensions/ProductExtensions.cs [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+                    product.Description = product.Description
+                        .Remove(match.Index, match.Length)
+                        .Insert(match.Index, $"<{tag}>{match.Value}</{tag}>");
+                }
+            }
+
+            return product;
+        }
+
+        internal static List<string> CollectCommonWords(this List<Product> products)
```
Comment (Admiraculix) [inline src/Warehouse/Application/Extensions/ProductExtensions.cs]: This is extension under List this means should be a separate file ListProductExtension

File: src/Warehouse/Application/Features/Products/Query/GetProductsHandler.cs [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    public class GetProductsHandler : IRequestHandler<GetProductsQuery, GetProductsResponse>
+    {
+        private readonly ICacheProvider _cacheProvider;
+
+        public GetProductsHandler(ICacheProvider cacheProvider)
+        {
+            _cacheProvider = cacheProvider;
+        }
+
+        public Task<GetProductsResponse> Handle(GetProductsQuery request, CancellationToken cancellationToken)
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Products/Query/GetProductsHandler.cs]: Why not have "async"?

File: src/Warehouse/Application/Features/Products/Query/GetProductsHandler.cs [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            var cachedProducts = _cacheProvider.Get<List<Product>>(AppConstants.ProductsCacheKey);
+            var summary = _cacheProvider.Get<ProductsSummary>(AppConstants.ProductsSummaryCacheKey);
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Products/Query/GetProductsHandler.cs]: lets make our project to be async...

File: src/Warehouse/Application/Features/Products/Query/GetProductsHandler.cs [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+
+            var filteredProducts = cachedProducts.Where(p =>
+                    (!request.MinPrice.HasValue || p.Price >= request.MinPrice)
+                    && (!request.MaxPrice.HasValue || p.Price <= request.MaxPrice)
+                    && (string.IsNullOrEmpty(request.Size) || p.Sizes.Contains(request.Size, StringComparer.OrdinalIgnoreCase)))
+                .Select(p => p.Highlight(request.Highlights));
+
+            return Task.FromResult(new GetProductsResponse()
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Products/Query/GetProductsHandler.cs]: You should automap here, not use manual map.

File: src/Warehouse/Application/Features/Products/Query/GetProductsResponse.cs [hunk 17]
```diff
@@ -0,0 +1,11 @@
+﻿using Domain.Models;
+
+namespace Application.Features.Products.Query
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Products/Query/GetProductsResponse.cs]: Lets move this into folder Models/Dtos

File: src/Warehouse/Application/Features/Products/Query/GetProductsResponse.cs [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+{
+    public class GetProductsResponse
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Products/Query/GetProductsResponse.cs]: This is GetProductDto or ProductDto.
Lets dtos in our business logic have suffix  Dto
and in our presentation layer(api) dtos have only suffix Request/Response.

File: src/Warehouse/Application/Validators/GetProductsQueryValidator.cs [hunk 19]
```diff
@@ -0,0 +1,23 @@
+﻿using Application.Features.Products.Query;
+using FluentValidation;
+
+namespace Application.Validators
+{
+    public class GetProductsQueryValidator : AbstractValidator<GetProductsQuery>
```
Comment (Admiraculix) [inline src/Warehouse/Application/Validators/GetProductsQueryValidator.cs]: The names speak that should be part of Query name space.

File: src/Warehouse/Application/Features/Products/Query/GetProductsQuery.cs [hunk 20]
```diff
@@ -0,0 +1,10 @@
+﻿using MediatR;
+
+namespace Application.Features.Products.Query
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Products/Query/GetProductsQuery.cs]: This is what I would recommend as a proper approach, how to structure on a feature basis in clean architecture:
image

File: src/Warehouse/Application/Features/Products/Synchronize/SyncProductsCommand.cs [hunk 21]
```diff
@@ -0,0 +1,6 @@
+﻿using MediatR;
+
+namespace Application.Features.Products.Synchronize
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Products/Synchronize/SyncProductsCommand.cs]: This is what I would recommend as a proper approach, how to structure on a feature basis in clean architecture:
image

This should be in folder: Commands/SyncProducts

File: src/Warehouse/Application/Features/Products/Synchronize/SyncProductsHandler.cs [hunk 22]
```diff
@@ -0,0 +1,38 @@
+﻿using Application.Extensions;
+using Domain.Models;
+using Infrastructure.Cache;
+using Infrastructure.Http;
+using MediatR;
+
+namespace Application.Features.Products.Synchronize
+{
+    public class SyncProductsHandler : IRequestHandler<SyncProductsCommand>
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Products/Synchronize/SyncProductsHandler.cs]: SyncProductsHandler -> SyncProductsCommandHandler

**Review by Admiraculix — CHANGES_REQUESTED** [review]

File: src/Warehouse/Application/Background/ProductsSyncService.cs (hunk 12 above)
Comment (minccccc) [reply src/Warehouse/Application/Background/ProductsSyncService.cs]: To be honest I do not log here, because I want (and I do) rely on the Logging pipeline behavior. I have a common approach for all the handlers - if there is a error inside to log the error into the logs.

What do you think?

File: src/Warehouse/Application/Features/Queries/GetProducts/GetProductsQueryValidator.cs [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Application.Features.Queries.GetProducts;
+
+public class GetProductsQueryValidator : AbstractValidator<GetProductsQuery>
+{
+    public GetProductsQueryValidator()
+    {
+        RuleFor(p => p.MinPrice)
+            .Must(m => m == null || m >= 0)
+            .WithMessage("MinPrice can not be negative");
```
Comment (Admiraculix) [inline src/Warehouse/Application/Features/Queries/GetProducts/GetProductsQueryValidator.cs]: If you decide, like other colleagues have done, you can also have validation messages in constants. The choice is yours.

File: src/Warehouse/Application/Models/GetProductsDto.cs [hunk 24]
```diff
@@ -0,0 +1,10 @@
+﻿using Domain.Models;
+
+namespace Application.Models;
```
Comment (Admiraculix) [inline src/Warehouse/Application/Models/GetProductsDto.cs]: for now it's good, you only have 1 model, but it would be great if there was a separate namespace/folder for the specific type of model:
Models/GetProductsDto.cs -> Models/DTOs/GetProductsDto.cs

File: src/Warehouse/Application/Services/Background/ProductsSyncService.cs [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+        _productsSourceConfig = productsSourcs.Value;
+    }
+
+    public async Task Execute(IJobExecutionContext context)
+    {
+        try
+        {
+            await _mediator.Send(new SyncProductsCommand(_productsSourceConfig.Uri));
+        }
+        catch { }
```
Comment (Admiraculix) [inline src/Warehouse/Application/Services/Background/ProductsSyncService.cs]: What do you think? Would it be nice to put some log in this match? Do not sit empty or rethrow..

File: src/Warehouse/Infrastructure/Cache/ICacheProvider.cs [hunk 26]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Infrastructure.Cache;
+
+public interface ICacheProvider
```
Comment (Admiraculix) [inline src/Warehouse/Infrastructure/Cache/ICacheProvider.cs]: image
Here you have broken the concept of Onion or Clean architecture. Currently, your application depends on an external layer of infrastructure. This is not right.
image


To fix it, create a folder in application: Interfaces,
all the interfaces you have in the infrastructure must go to the Application and delete the reference to the infrastructure assembly.

File: src/Warehouse/Infrastructure/Cache/InMemoryCacheProvider.cs [hunk 27]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.Extensions.Caching.Memory;
+
+namespace Infrastructure.Cache;
+
+public class InMemoryCacheProvider : ICacheProvider
```
Comment (Admiraculix) [inline src/Warehouse/Infrastructure/Cache/InMemoryCacheProvider.cs:6]: Broken the concept of Onion or Clean architecture. 
Check previous commentary.

File: src/Warehouse/Infrastructure/Http/IRetrieveProductsService.cs [hunk 28]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Infrastructure.Http;
+
+public interface IRetrieveProductsService
```
Comment (Admiraculix) [inline src/Warehouse/Infrastructure/Http/IRetrieveProductsService.cs]: Broken the concept of Onion or Clean architecture. 
Check previous commentary.

File: src/Warehouse/Infrastructure/Http/RetrieveProductsService.cs [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+
+                return JsonConvert.DeserializeObject<T>(stringResult);
+            }
+            else
+            {
+                throw new Exception($"Failed to retrieve products from the source {response.StatusCode}");
+            }
+        }
+        catch (Exception)
+        {
```
Comment (Admiraculix) [inline src/Warehouse/Infrastructure/Http/RetrieveProductsService.cs]: Should log here

File: src/Warehouse/Infrastructure/Http/RetrieveProductsService.cs [hunk 30]
```diff
@@ -0,0 +1,39 @@
+﻿using Newtonsoft.Json;
+
+namespace Infrastructure.Http;
+
+public class RetrieveProductsService : IRetrieveProductsService
```
Comment (Admiraculix) [inline src/Warehouse/Infrastructure/Http/RetrieveProductsService.cs]: I believe that code should reflect the real world. Since we use OOP as a programming paradigm, names are an important part of our code.
It recommends changing the name, which I think is more descriptive than the current one:
 IMockyHttpClientService.cs

File: src/Warehouse/WebApi/Controllers/StockController.cs [hunk 31]
```diff
@@ -0,0 +1,34 @@
...
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using WebApi.Models.Requests;
+using WebApi.Models.Responses;
+
+namespace WebApi.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class StockController : ControllerBase
```
Comment (Admiraculix) [inline src/Warehouse/WebApi/Controllers/StockController.cs]: Create your base controller that inherits ControllerBase from .netcore. Yours should take all common things and be inherited by Stock controller.
The idea is that if you add a new controller which is a lot of you probably inherit your controller and save all the boiler code:
[ApiController]
[Route("api/[controller]")] and
constructor with mediator and mapper
or
The other approach is from new core version is FromService:
https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/dependency-injection?view=aspnetcore-8.0#action-injection-with-fromservices

File: src/Warehouse/WebApi/appsettings.json [hunk 32]
```diff
@@ -0,0 +1,24 @@
...
+    "WriteTo": [
+      { "Name": "Console" }
+    ],
+    "Enrich": [ "FromLogContext", "WithMachineName", "WithThreadId" ],
+    "Properties": {
+      "Application": "Warehouse"
+    }
+  },
+  "AllowedHosts": "*",
+  "ProductsSource": {
```
Comment (Admiraculix) [inline src/Warehouse/WebApi/appsettings.json]: This name is not good 
I think the name is very abstract and it is not clear what it is used for:
```
 "MockyClient": {
    "BaseUrl": "http://www.mocky.io/",
    "ProductUrl": "v2/5e307edf3200005d00858b49"
  }
```

File: src/Warehouse/WebApi/appsettings.json [hunk 33]
```diff
@@ -0,0 +1,24 @@
...
+    "Uri": "https://run.mocky.io/v3/97aa328f-6f5d-458a-9fa4-55fe58eaacc9",
+    "RefreshTime": 10
```
Comment (Admiraculix) [inline src/Warehouse/WebApi/appsettings.json]: this is not related for Http client.
This mean is related with your job.
Should be speared object like that or similar to this:
```
 "Scheduler": {
    "RecurringJobTimeInMinutes": 10,
  }
```

File: src/Warehouse/WebApi/appsettings.json [hunk 34]
```diff
@@ -0,0 +1,24 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft.AspNetCore": "Warning"
+    }
+  },
+  "Serilog": {
```
Comment (Admiraculix) [inline src/Warehouse/WebApi/appsettings.json]: Line 9 to line 24 in my opinion should be moved into: appsettings.Development.json.

Because if you think about it,
and you have 3 environments(development, stage, production), these settings are for the local one, namely develop.
i.e. are not common.

File: src/Warehouse/WebApi/MappingProfiles/ProductMappingProfiles.cs [hunk 35]
```diff
@@ -0,0 +1,21 @@
...
+using AutoMapper;
+using Domain.Models;
+using WebApi.Models.Requests;
+using WebApi.Models.Responses;
+
+namespace WebApi.MappingProfiles;
+
+public class ProductMappingProfiles : Profile
+{
+    public ProductMappingProfiles()
```
Comment (Admiraculix) [inline src/Warehouse/WebApi/MappingProfiles/ProductMappingProfiles.cs:12]: There are two options for how you can do it more organized here.
1. The approach is to map 2 private methods to represent the mapping direction.
```
    public MappingProfile()
    {
        MapFromRequestToQueriesOrCommands();
        MapFromDtoToResponse();
    }

    private void MapFromRequestToQueriesOrCommands()
    {
        CreateMap<ProductFilterRequest, ProductQuery>();
    }

    private void MapFromDtoToResponse()
    {
        CreateMap<ProductDto, ProductResponse>();
    }
```
2. The approach is to have 2 profiles with each direction

```
public class MapFromRequestToQueriesOrCommands : Profile { }
public class MapFromDtoToResponse : Profile {}
```
Decide what you like best and use the approach.

This helps a person to quickly orient himself in the complex structure of the project and to expand the maps more easily

File: tests/UnitTests/ArchitectureTests.cs [hunk 36]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    [Fact]
+    public void Application_Should_Not_HaveDependencyOnOtherProjects()
+    {
+        var assembly = typeof(Application.Extensions.DependencyRegistrationExtention).Assembly;
+
+        var otherProjects = new[]
+        {
+            InfrastructureNamespace,
```
Comment (Admiraculix) [inline tests/UnitTests/ArchitectureTests.cs]: This test is correct as an implementation BUT is not correct as a result.
In the current implementation, you HAVE project reference to infrastructure.:
image

Maybe there are some bugs in Nugget. But should investigate what is going on.

File: tests/UnitTests/Mocks/ICacheProviderMock.cs [hunk 37]
```diff
@@ -0,0 +1,83 @@
+﻿using Application.Common.Constants;
+using Domain.Models;
+using Infrastructure.Cache;
+using Moq;
+
+namespace UnitTests.Mocks;
+
+public static class ICacheProviderMock
```
Comment (Admiraculix) [inline tests/UnitTests/Mocks/ICacheProviderMock.cs]: I haven't come across this approach, but I don't think static classes should start with "I" like interfaces. 
This applies to all your mocks static classes.

File: tests/UnitTests/UnitTests.csproj [hunk 38]
```diff
@@ -0,0 +1,35 @@
...
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+    <IsTestProject>true</IsTestProject>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="FluentAssertions" Version="6.12.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.6.0" />
+    <PackageReference Include="Moq" Version="4.20.70" />
```
Comment (Admiraculix) [inline tests/UnitTests/UnitTests.csproj]: New versions of Moq are not welcome because they sell information.
https://www.youtube.com/watch?v=A06nNjBKV7I&ab_channel=NickChapsas

File: tests/UnitTests/Features/GetProductsHandlerTests.cs [hunk 39]
```diff
@@ -0,0 +1,99 @@
+using Application.Common.Constants;
+using Application.Features.Queries.GetProducts;
+using Application.Models;
+using AutoMapper;
+using Domain.Models;
+using FluentAssertions;
+using Infrastructure.Cache;
+using Microsoft.OpenApi.Any;
```
Comment (Admiraculix) [inline tests/UnitTests/Features/GetProductsHandlerTests.cs]: Clear all not used usings. This comment applies to all files

File: tests/UnitTests/Features/GetProductsHandlerTests.cs [hunk 40]
```diff
@@ -0,0 +1,99 @@
...
+
+namespace UnitTests.Features;
+
+public class GetProductsHandlerTests
+{
+    [Fact]
+    public async Task Should_ReturnAllProducts()
+    {
+        var query = new GetProductsQuery(null, null, new List<string>(), null);
+        var result = await SendToHandler(query);
```
Comment (Admiraculix) [inline tests/UnitTests/Features/GetProductsHandlerTests.cs:19]: Let's apply the AAA pattern.

```
//Arrange
	Setup mocks

//Act
	Call the system under test (SUT) 
	[The System Under Test (SUT) from a Unit Testing perspective represents all of the actors (i.e. one or more classes) in a test that are not mocks or stubs.]

//Assert
	Assertion phase
```

This  applies to all tests

File: tests/UnitTests/Features/GetProductsHandlerTests.cs [hunk 41]
```diff
@@ -0,0 +1,99 @@
...
+            }
+        }
+    }
+
+    private async Task<GetProductsDto> SendToHandler(GetProductsQuery query)
+    {
+        var cacheProviderMock = ICacheProviderMock.GetMock();
+        var mapperMock = IMapperMock.GetMock();
+
+        var handler = new GetProductsQueryHandler(cacheProviderMock.Object, mapperMock.Object);
```
Comment (Admiraculix) [inline tests/UnitTests/Features/GetProductsHandlerTests.cs]: this is you sut:
var handler -> var sut

**Review by Admiraculix — CHANGES_REQUESTED** [review]
